patches/0029-Handle-cases-where-region-exists-and-is-nil.patch

```diff
@@ -0,0 +1,21 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: corymhall <43035978+corymhall@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:24:02 -0400
+Subject: [PATCH] Handle cases where region exists and is nil
+
+
+diff --git a/internal/provider/provider.go b/internal/provider/provider.go
+index 8512e5dfde..f4359bad72 100644
+--- a/internal/provider/provider.go
++++ b/internal/provider/provider.go
+@@ -533,7 +533,9 @@ func initialize(ctx context.Context, provider *schema.Provider) (map[string]conn
+ 
+ 					if isRegionOverrideEnabled && getAttribute != nil {
+ 						if region, ok := getAttribute(names.AttrRegion); ok {
+-							overrideRegion = region.(string)
++							if val, ok := region.(string); ok {
```

Comment:
TODO: once merged create tracking issue. Maybe a bridge issue.

Reply:
What's `overrideRegion` prior to this line? The remaining code is in another patch, right?

Reply:
The rest of the code is just upstream code. The only patch we have in this code is what i've added here. 

The line above which isn't included in the patch just initializes overrideRegion. 

```go
var overrideRegion string
```

`getAttribute` is a function looks up the attribute in the rawState, essentially something like this:

```go
getAttribute := func(key: string): (any, bool) {
  v, ok := rawState[key]; return v, ok
}
```

In Terraform it looks like at the point this code is running `rawState` does not have a `region` attribute at all so `getAttribute` returns `false`. In Pulumi for some reason `rawState` does have a `region` attribute with a `nil` value. I think it's a bridge issue so I'll create a follow up issue.